File: Programmers/위클리챌린지/Lv1부족한금액계산하기.java
public class Lv1부족한금액계산하기 {

    public long solution(int price, int money, int count) {
        long answer = -money;

        for (int i = 1; i <= count; i++) {
            answer += (long) price * i;
        }
        if (answer < 0L) {
            answer = 0L;
        }

        return answer;
    }

}

File: Programmers/위클리챌린지/Lv1부족한금액계산하기.kt
class Solution {
    fun solution(price: Int, money: Int, count: Int): Long {
        val value: Long = (1L .. count).asSequence()
            .map { it * price }
            .sum() - money

        if (value < 0L) {
            return 0L
        }
        return value
    }
}